ceasiompy/ActuatorDisk/func/optimalprop.py

```diff
@@ -0,0 +1,416 @@
+    file.write("NROW= " + str(stations) + "\n")
+    file.write("# rs=r/R        dCT/drs       dCP/drs       dCR/drs\n")
+    for r, ctrs, cprs in zip(r, CTrs, CPrs):
+        file.write(f"  {r:.7f}     {ctrs:.7f}     {cprs:.7f}     0.0\n")
+
+    file.close()
+
+    log.info("Warning: present version requires input in SI units.")
+
+
+def thrust_calculator(
+    stations,
+    total_thrust_coefficient,
+    radius,
+    hub_radius,
+    advanced_ratio,
+    free_stream_velocity,
+    prandtl,
+    blades_number,
+):
+
+    """Performing of a calculation to obtain thrust and power coefficients distribution
+
+    Args:
+            stations (float): Number of elements for blade discretization [-]
+            total_thrust_coefficient (float): Total thrust coefficient[-]
+            radius (float): Blade radius [m]
+            hub_radius (float): Radius of the blade at the hub [m]
+            advanced_ratio (float): Free_stream_velocity/(rotational_velocity*diameter)[-]
+            free_stream_velocity (float): Cruise velocity [m/s]
+            prandtl (bool): Correction for tip losses
+            blades_number (int): Blades propeller number[-]
+
+    Returns:
+        dCt_optimal (float): thrust coefficient at every radius [-]
+        dCp (float): power coefficient at every radius [-]
+        r (float): adimensional radius [-]
+    """
+
+    # Resize the vectors using the number of radial stations.
+    r = np.empty(stations)
+    non_dimensional_radius = np.empty(stations)
+    dCp = np.empty(stations)
+    induced_velocity_distribution = np.empty(stations)
+    new_axial_interference_factor = np.empty(stations)
+    old_axial_interference_factor = np.empty(stations)
+    initial_axial_interference_factor = np.empty(stations)
+    dCt_new = np.empty(stations)
+    dCt_old = np.empty(stations)
+    dCt_0 = np.empty(stations)
+    delta_pressure = np.empty(stations)
+    correction_function = np.empty(stations)
+    optimal_axial_interference_factor = np.empty(stations)
+    optimal_rotational_interference_factor = np.empty(stations)
+    dCt_optimal = np.empty(stations)
+
+    non_dimensional_hub_radius = hub_radius / radius
+
+    # Computation of the non-dimensional radial stations.
+    for i in range(1, stations + 1):
+        r[i - 1] = i / float(stations)
+        if r[i - 1] <= non_dimensional_hub_radius:
+            i_hub = i - 1
+
+    diameter = 2 * radius
+    n = free_stream_velocity / (diameter * advanced_ratio)
+    omega = n * 2 * pi
+
+    # Computation of the tip loss Prandtl correction function
+    if prandtl:
+        for i in range(stations):
+            correction_function[i] = (2 / pi) * acos(
+                exp(
+                    -0.5
+                    * blades_number
+                    * (1 - r[i])
+                    * sqrt(1 + (omega * radius / free_stream_velocity) ** 2)
+                )
+            )
+
+    else:
+        correction_function[stations] = 1.0
```

Comment:
If it is is not prandtl you will only change the `correction_function` for the last value of the list, is it correct?